core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -192,7 +192,12 @@ class BaseColumnFormatRelation(
       truncate()
     }
     JdbcExtendedUtils.saveTable(data, table, schema, connProperties)
-    flushRowBuffer()
+
+    SnappyContext.getClusterMode(_context.sparkContext) match {
```

Comment:
Low priority comment: This can be moved to ExternalStoreUtils, just like isSplitOrLocalMode.